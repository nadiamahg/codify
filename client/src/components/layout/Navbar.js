import React, { Component } from "react";
import { Link } from "react-router-dom";
import logo from '../../images/logo.png';
import PropTypes from "prop-types";
import { connect } from "react-redux";
import { logoutUser } from "../../actions/authActions";
class Navbar extends Component {

  constructor(props) {
    super(props);
    this.state = {
      username: null,
      isLoading: true,
    }
  };

  getUsername = function(){
    return this.props.auth.user.username;
  }

  onLogoutClick = e => {
    e.preventDefault();
    this.setState({
        username: null
    });
    this.props.logoutUser();
  };

  render() {
    return (
      <div className="navbar-fixed">
        <nav className="z-depth-0">
          <div className="nav-wrapper blue darken-3">
            <Link
              to="/"
              style={{
                fontFamily: "monospace",
                paddingLeft: "20px",
              }}
              className="col s5 brand-logo white-text"
            >
              <i style={{paddingTop: "6px"}}><img src={logo} alt="Logo"/></i>
              CODIFY
            </Link>
           <div className="right">{this.getUsername()} {!this.getUsername() ? "" : 

            <Link
              to="/"
              className="btn btn-small waves-effect waves-light hoverable blue accent-3"
              onClick={this.onLogoutClick}
            >
              Log Out
            </Link>


           }
           </div>
          </div>
        </nav>
      </div>
    );
  }

}

Navbar.propTypes = {
  logoutUser: PropTypes.func.isRequired,
  auth: PropTypes.object.isRequired
};
const mapStateToProps = state => ({
  auth: state.auth
});
export default connect(
  mapStateToProps, { logoutUser })(Navbar);