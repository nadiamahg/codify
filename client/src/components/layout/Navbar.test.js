import React from 'react';
import renderer from 'react-test-renderer';
import Navbar from './Navbar';
import { BrowserRouter} from 'react-router-dom';
import Adapter from 'enzyme-adapter-react-16';
import { shallow, configure } from 'enzyme';

configure({adapter: new Adapter()});
const initialState = {
  auth: {}
};

const props = {
	auth: {
		user: {
			username: "Nadia"
		}
	}
}

import configureMockStore from 'redux-mock-store';
import { Provider } from 'react-redux'
const mockStore = configureMockStore();
const store = mockStore(initialState);


describe('Test suits for Navbar', () => {
it('renders', () => {
	const wrapper = shallow(<Navbar store={store} {...props}/>);
	expect(wrapper).toMatchSnapshot();

 });
});
