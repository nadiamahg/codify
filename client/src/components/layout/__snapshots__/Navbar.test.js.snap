// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test suits for Navbar renders 1`] = `ShallowWrapper {}`;
